feat: use published bifold

jleach opened the pull request:
Adapt BC Wallet to use the `@hyperledger/aries-bifold-core` packages published to NPM.

cvarjao committed:
switch to linked dependency

Signed-off-by: Clécio Varjão <1348549+cvarjao@users.noreply.github.com>

cvarjao committed:
refactor to use linked packages

Signed-off-by: Clécio Varjão <1348549+cvarjao@users.noreply.github.com>

cvarjao committed:
update lock files

Signed-off-by: Clécio Varjão <1348549+cvarjao@users.noreply.github.com>

jleach committed:
chore: update after rebase

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
fix: dependencies

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
fix: update path in ledger update action

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
fix: update yarn lock

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

cvarjao commented on package.json:
```diff
@@ -59,7 +56,6 @@
     "@aries-framework/react-native@0.4.0": "patch:@aries-framework/react-native@npm:0.3.3#./.yarn/patches/@aries-framework-react-native-npm-0.3.3-bb75ece22d.patch",
     "@aries-framework/anoncreds@^0.4.0": "patch:@aries-framework/anoncreds@npm%3A0.4.0#./.yarn/patches/@aries-framework-anoncreds-npm-0.4.0-4d3b4e769d.patch",
-    "@aries-framework/anoncreds@0.4.0": "patch:@aries-framework/anoncreds@npm%3A0.4.0#./.yarn/patches/@aries-framework-anoncreds-npm-0.4.0-4d3b4e769d.patch",
-    "@aries-framework/indy-vdr@^0.4.0": "patch:@aries-framework/indy-vdr@npm%3A0.4.0#./.yarn/patches/@aries-framework-indy-vdr-npm-0.4.0-b01067562c.patch"
```
is this being removed by accident?

jleach replied:
I'm not sure what happened here. I still see `@aries-framework/anoncreds@0.4.0` and added VDR back.

wadeking98 replied:
there were two anoncreds patches here, I believe one was the patch to the revocation interval that jason made, The other was a speed improvement from me when looking up the revocation registry for multiple credentials. Have the patches been combined or was one accidentally deleted?

bryce-mcmath replied:
they appear to have been duplicates, looking at the hash

jleach committed:
fix: lock AFJ pkg version

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
fix: add vdp patch back

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
fix: missing import

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

bryce-mcmath commented on package.json:
```diff
@@ -55,6 +52,13 @@
+    "@aries-framework/react-native": "0.4.0",
     "@aries-framework/core@0.3.3": "patch:@aries-framework/core@npm:0.3.3#./.yarn/patches/@aries-framework-core-npm-0.3.3-dd6486de3d.patch",
     "@aries-framework/react-hooks@^0.4.2": "patch:@aries-framework/react-hooks@npm:0.4.2#./.yarn/patches/@aries-framework-react-hooks-npm-0.4.2-84b7eb8764.patch",
     "@aries-framework/react-native@0.4.0": "patch:@aries-framework/react-native@npm:0.3.3#./.yarn/patches/@aries-framework-react-native-npm-0.3.3-bb75ece22d.patch",
```
Is there no issue with having two resolutions for different versions of the same dep? ie. core 0.3.3 vs core 0.4.0, react-native 0.3.3 vs 0.4.0

cvarjao replied:
Agreed. line 58 (`"@aries-framework/core": "0.4.0"`) and line 62 (`"@aries-framework/core@0.3.3": "patch:@..."`) are conflicting with each other. That means the patch won't apply to core 0.4.0

jleach replied:
Ok. I'll remove 0.3.3 since we don't use it. I though those lines meant that if you were installing v0.3.3 do one thing, if you were installing 0.4.0 do something different.

jleach committed:
fix: cleanup patches

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

cvarjao committed:
Resolve AFJ dependency mimatches

# delete all node_modules

find . -name 'node_modules' -type d -prune -exec rm -rf '{}' +

# reset lock file from main

git restore --source origin/main -- yarn.lock

# install dependencies

yarn install

Signed-off-by: Clécio Varjão <1348549+cvarjao@users.noreply.github.com>

jleach committed:
chore: remove bifold submodule

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
chore: use yarn 4.0

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
fix: metro use attestation mod

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
fix: use checkout for bifold

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
fix: use checkout for bifold

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
fix: fix secrets

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
fix: update yarn lock

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

jleach committed:
chore: remote update ledgers

Signed-off-by: Jason C. Leach <jason.leach@fullboar.ca>

wadeking98 commented:
Other than my one comment LGTM
